spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/DynamicInterpreter.scala

```diff
@@ -110,6 +120,27 @@ object DynamicInterpreter {
     optimizerTimeTracker.cleanupWorkaroundDf(conformedDf)
   }
 
+  private def checkMutabilityNotViolated(schema: StructType, steps: List[ConformanceRule])
```

Comment:
*Rant/thought:* I see a basis for Conformance rules validation here. An API, that the UI would call to indicate that the set of rules is:
* executable
* warn it might not work (based on configuration)
* not correct at all

The whole thing is out of scope of this task, of course. But would be nice to have it in mind 😉

Reply:
We are now talking UI validations. We even have an issue for it. 

If we validate on creation, there is no need for call from spark-jobs to check

Reply:
My point-of-view it will always need to be done before executing Conformance. The UI can be bypassed, the rules behavior changed or schema altered.
But as mentioned before out of scope of this. 🐱